daemon/SecretsImpl/pluginfunctionwrappers.cpp

```diff
@@ -18,6 +18,33 @@ PluginState Daemon::ApiImpl::pluginState(PluginBase *plugin)
     return PluginState(plugin->isAvailable(), plugin->isLocked());
 }
 
+FoundLockStatusResult Daemon::ApiImpl::queryLockSpecificPlugin(
+        const QMap<QString, Sailfish::Secrets::EncryptionPlugin*> &encryptionPlugins,
+        const QMap<QString, StoragePluginWrapper*> &storagePlugins,
+        const QMap<QString, EncryptedStoragePluginWrapper*> &encryptedStoragePlugins,
+        const QString &lockCodeTarget)
+{
+    bool found = true;
+    Sailfish::Secrets::LockCodeRequest::LockStatus lockStatus = Sailfish::Secrets::LockCodeRequest::Unknown;
+    Result result(Result::Succeeded);
+    PluginBase *p = storagePlugins.contains(lockCodeTarget) ? static_cast<PluginBase*>(storagePlugins.value(lockCodeTarget))
```

Comment:
This is a bit unfortunate contains + value. Passing default value to the value simplifies this and avoid extra lookup.

E.g. static_cast<PluginBase*>(storagePlugins.value(lockCodeTarget, nullptr))

Reply:
In this ternary case I don't think it does simplify, as we need to change the assignment if the value is null (i.e. if the map does not contain the value).  I could use chained ifs to get the same behaviour with better performance, so I will change it to that.